src/Security/SecSharedCredential.cs

```diff
@@ -87,13 +107,24 @@ static internal class ArrayErrorActionTrampoline {
 			}
 		}
 
+#if !XAMCORE_4_0
+		[Obsolete ("Use the overload accepting a 'SecSharedCredentialInfo' argument.")]
 		public static void RequestSharedWebCredential (string domainName, string account, Action<string[], NSError> handler)
 		{
-			// do not check domain an account because they can be null
+			throw new InvalidOperationException ("Use correct delegate type");
+		}
+#endif
+
+		[iOS (8,0)]
+		public static void RequestSharedWebCredential (string domainName, string account, Action<SecSharedCredentialInfo[], NSError> handler)
+		{
 			Action<NSArray, NSError> onComplete = (NSArray a, NSError e) => {
-				// get a string [] for the user rather than an ugly NSArray
-				var array = NSArray.StringArrayFromHandle (a.Handle);
-				handler (array, e);
+				var creds = new SecSharedCredentialInfo [a.Count];
+				for (xint i = 0; i < a.Count; i++) {
```

Comment:
Minor: `a.Count` requires a trip to native code, using `creds.Count` instead would avoid that.